src/app.rs

```diff
@@ -233,7 +231,7 @@ impl Application {
                         // Reload paths
                         config_paths = config::process_paths(&opts.config_paths_with_formats()).unwrap_or(config_paths);
                         // Reload config
-                        let new_config = config::load_from_paths(&config_paths, false).map_err(handle_config_errors).ok();
+                        let new_config = config::load_from_paths(&config_paths).map_err(handle_config_errors).ok();
```

Comment:
Don't we need to do the same preinit here in-case the globals changed? Is it possible to move the global log schema initialization into `config::load_from_paths`? I think that would avoid neglecting to call it together each time.

Reply:
> Don't we need to do the same preinit here in-case the globals changed?

Nope. We can only set global schema once, that's also the reason those two steps/functions are separated.

Reply:
I'm not sure I completely understand. If I had a config file like:

```toml
host_key = "foo"

// source, sinks, etc.
```

and then I modified it to:

```toml
host_key = "bar"

// source, sinks, etc.
```

Wouldn't we need to redo just the global config again before reinitializing the components so that they pick up the new default?

Reply:
> Wouldn't we need to redo just the global config again before reinitializing the components so that they pick up the new default?

It's more complicated:
* we would need to change global `LOG_SCHEMA` variable to be changeable
* we would need to shutdown and rebuild even unchanged components. This is because when changing to a new configuration we compute the difference between topologies, retain the common part, shutdown removed, add new, and recreate changed components.

For similar reason `data_dir` can't be changed and we fail reloading if it's different from initial value, which we should also do for log schema which I've checked now and it seams we aren't checking for that so I'll add it.

Reply:
More generally any global option related to components can't be changed.

Reply:
Ah, interesting, that is a limitation I was now aware of. It seems like we could permit things like `data_dir` to be changed with the caveat that, if users change this, vector will forget whatever its previous state was.

In my opinion, we should let users change and reload global config values as well for consistency. But if we don't want to do that for now, I think it'd be good to explicitly fail the reload if global values were changed. Are we doing that?

Reply:
> In my opinion, we should let users change and reload global config values as well

It's doable by just rebuilding everything, and I agree it would be nice to cover that case so could you open an issue for that.

>  It seems like we could permit things like data_dir to be changed with the caveat that, if users change this, vector will forget whatever its previous state was.

We are flushing everything that is being removed by default so that won't be an issue.

> , I think it'd be good to explicitly fail the reload if global values were changed. Are we doing that?

I agree, and no we aren't so I'll add a check for that.

Reply:
👍 thanks for the discussion!

I opened https://github.com/timberio/vector/issues/6598 to track enhancing the reload to allow for changing global options.

Thanks for adding a check to explicitly fail. I think this will be better behavior than silently ignoring changes to these values.